lib/md4.c

```diff
@@ -93,10 +93,10 @@
 
 typedef struct md4_ctx MD4_CTX;
 
-static CURLcode MD4_Init(MD4_CTX *ctx)
+static int MD4_Init(MD4_CTX *ctx)
 {
   md4_init(ctx);
-  return CURLE_OK;
+  return 1;
```

Comment:
@danielgustafsson I've changed MD4_Init prototype because it has to match backends that provide that function such as OpenSSL. [See here](https://www.openssl.org/docs/manmaster/man3/MD4_Init.html)

Another way to do it would be the way we do for md5 and sha256 where for every backend we define a CURLcode my_md4_init that either wraps MD4_Init or does whatever the alternative MD4 init procedure.

Also, I am not sure about this one above, I cannot find anything about GnuTLS md4_init in documentation. If it returns a code then this will need to be changed.

Reply:
GnuTLS uses nettle for crypto. nettle's md4_init function does not return anything:

https://git.lysator.liu.se/nettle/nettle/-/blob/master/md4.c#L58-74